incubator/hnc/internal/validators/object.go

```diff
@@ -139,6 +143,30 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	return o.handleInherited(op, newSource, oldSource, inst, oldInst)
 }
 
+func validateSelectorAnnot(inst *unstructured.Unstructured) string {
+	annots := inst.GetAnnotations()
+	for key, _ := range annots {
+		// for example: segs = ["propagate.hnc.x-k8s.io", "select"]
+		segs := strings.SplitN(key, "/", 2)
+		// for example: prefix = ["propagate", "hnc.x-k8s.io"]
+		prefix := strings.SplitN(segs[0], ".", 2)
+		if prefix[1] != api.MetaGroup {
+			continue
+		}
+		// If this annotation is part of HNC metagroup, we check if the prefix value is valid
+		if segs[0] != api.AnnotationPropagatePrefix {
+			return fmt.Sprintf("invalid HNC annotation: %v", key)
+		}
+		// check if the suffix is valid by checking the whole annotation key
+		if key != api.AnnotationSelector &&
+			key != api.AnnotationTreeSelector &&
+			key != api.AnnotationNoneSelector {
+			return fmt.Sprintf("invalid HNC annotation: %v", key)
```

Comment:
I think it would be more user-friendly to print out the valid options here.

Reply:
I thought we were only checking the exceptions annotation here, so I'll only print valid exceptions annotations, is that correct?